packages/@uppy/companion-client/src/Provider.ts

```diff
@@ -58,48 +114,63 @@ export default class Provider extends RequestClient {
     return { ...headers, ...authHeaders }
   }
 
-  onReceiveResponse(response) {
+  onReceiveResponse(response: Response): Response {
     super.onReceiveResponse(response)
-    const plugin = this.uppy.getPlugin(this.pluginId)
+    const plugin = this.#getPlugin()
     const oldAuthenticated = plugin.getPluginState().authenticated
-    const authenticated = oldAuthenticated ? response.status !== authErrorStatusCode : response.status < 400
+    const authenticated = oldAuthenticated
+      ? response.status !== authErrorStatusCode
+      : response.status < 400
     plugin.setPluginState({ authenticated })
     return response
   }
 
-  async setAuthToken(token) {
-    return this.uppy.getPlugin(this.pluginId).storage.setItem(this.tokenKey, token)
+  async setAuthToken(token: string): Promise<void> {
+    return this.#getPlugin().storage.setItem(this.tokenKey, token)
   }
 
-  async #getAuthToken() {
-    return this.uppy.getPlugin(this.pluginId).storage.getItem(this.tokenKey)
+  async #getAuthToken(): Promise<string | null> {
+    return this.#getPlugin().storage.getItem(this.tokenKey)
   }
 
-  /** @protected */
-  async removeAuthToken() {
-    return this.uppy.getPlugin(this.pluginId).storage.removeItem(this.tokenKey)
+  protected async removeAuthToken(): Promise<void> {
+    return this.#getPlugin().storage.removeItem(this.tokenKey)
+  }
+
+  #getPlugin() {
+    const plugin = this.uppy.getPlugin(this.pluginId) as ProviderPlugin<M, B>
+    if (plugin == null) throw new Error('Plugin was nullish')
```

Comment:
We might consider adding an assertion util that would throw something like `Uppy internal assertion failed. This indicates a bug within Uppy – or unsupported use of Uppy internals. Consider opening an issue at https://github.com/transloadit/uppy/issues/new with the stack trace`? We might find ourselves needing this more and more as want stronger types

Reply:
For now I think it's fine to determine it case by case basis. Let's see how we feel once we implement more.